spec/services/datetime_parser_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe DatetimeParser do
+  describe "#parse" do
+    it "returns the parsed date/time when valid" do
+      date_params = { day: "10", month: "01", year: "2019" }
+      expected_time = Time.zone.local(2019, 1, 10, 0, 0)
+
+      parser = DatetimeParser.new(date: date_params,
+                                  time: "11:00am")
+      expect(parser.parse).to eq expected_time.change(hour: 11)
```

Comment:
Rather than using `.change` can you show what the actual date/time being set is, e.g. `Time.zone.local(2019, 1, 10, 11, 0)`.